---
title: "Geneset enrichment analysis"
editor_options: 
  chunk_output_type: console
fig-width: 12
fig-height: 10
---


## Pathway/gene set enrichment analysis

After differential gene expression (DGE) analysis we have a long list of differentially expressed genes. Pathway enrichment analysis (PEA) is a common approach to summarise this long list of genes and interpret hundreds of DGEs at once. So instead of having a list of 20 000 genes, we may get a list of 50 or 60 biological pathways.

![Summarise lists of DGEs to pathways](/images/GSE_analysis/pea.png)

Pathway enrichment analysis summarises the long gene list to a shorter and more easily interpretable list of pathways.

### How does PEA work?

For a pathway enrichment analysis we need 3 ingredients.

1.  Our **gene list** of interest, fx. a list of DGEs which we want to summarise.
2.  A list of **background genes** - fx all of the genes in the human genome/transcriptome.
3.  A list of **gene sets**. Gene sets are basically groups of related genes. Of course, for the algorithm to know if your list has a lot of genes related to breast cancer, or apoptosis, or cellular respiration, you need to tell it which genes are actually involved in breast cancer, apoptosis, and cellular respiration.

![Components of a pathway enrichment analysis](/images/GSE_analysis/pea_2.png)

PEA essentially compares our gene list to the background list to check if there are certain pathways **overrepresented**

One example could be: Alcoholic liver disease is usually involved in inflammatory processes, which often involve pro-inflammatory cytokines like IL-6. So, is there an **association** between our genes differentially expressed in alcoholic liver disease vs healthy cells and IL-6 production.

![](/images/GSE_analysis/pea_3.png)

Is our list of DGEs **enriched** with genes involved in IL-6 synthesis pathway?

![](/images/GSE_analysis/pea_4.png)

To answer this question, we can build a contigency table. This will help us determine whether the fraction of genes of interest in the pathway is higher compared to the fraction of genes outside the pathway (so, backgound set). See the table below.

![](/images/GSE_analysis/pea_5.png)

We have a column for differentially expressed and a column not differentially expressed genes, and then two rows, for genes that are annotated as being involved in IL-6 production and genes that are not involved in IL6 production.

To simplify things a lot, we will just look at 30 genes. 15 deferentially expressed genes were identified and of those, 12 genes were associated with the GO term interleukin-6 production.

We find that **12 of our 15** differentially expressed genes are involved in IL-6 production. We could quite confidently say that **our gene list is enriched with genes involved in IL-6 production.**

But what if there was 9 of our 15 involved in IL-6 production. is our gene list enriched with genes involved in IL-6 production? We need an objective statistical test to determine what is enriched and what is not. There are many methods, by the one that is commonly used in pathway enrichment analysis is fisher's exact test. I our p-value is really lwo we can safely say that our list is overrepresented with genes involved in IL-6 production, in other words, IL-6 prodcution is an immportant pathway in alchoholic liver diseases compared to healthy liver cells. Becayse we a testing thousands of pathways we till have to adjust p-values with a multiple-testing correction method.

Pathway enrichment analysis takes your **gene list** of interest and compares it to a **list of background genes** to check if there are certain pathways that are over-represented.

So it checks the fraction of your genes annotated to a specific Gene Ontology (GO) term. Then it checks the proportion of genes in the whole genome (your background set) that are annotated to that GO Term.

Then, it gives you a **p-value** which tells you what is the probability that that pathway is actually over-represented in your gene list and it wasn't just coincidence.

### Our analysis

Mark Burton have done the GSE analysis using four different databases, outlined below. I think we just use the vehicle control as the list of background genes. Otherwise he must have used some subset of only genes/pathways related to PHHs.

Gene Ontology (GO), KEGG, Reactome, and BioCarta are databases and tools that provide comprehensive information on gene and protein functions, interactions, and pathways. Each has its unique focus and strengths:

1.  **Gene Ontology (GO)**

    -   **Focus:** Provides a standardized vocabulary (ontologies) to describe gene and protein roles in cells. It covers three domains: Biological Process (BP), which describes the biological objectives to which the gene or gene product contributes; Molecular Function (MF), which describes the elemental activities of a gene product at the molecular level, such as binding or catalysis; and Cellular Component (CC), which describes where in the cell the gene product is active.

    -   **Strengths:** Widely used for annotating genes and gene products across species; facilitates consistent descriptions of gene products even in the context of different studies.

2.  **Kyoto Encyclopedia of Genes and Genomes (KEGG)**

    -   **Focus:** Integrates genomic, chemical, and systemic functional information. KEGG provides pathways maps that represent molecular interaction and reaction networks, covering metabolism, genetic information processing, environmental information processing, cellular processes, organismal systems, and human diseases.

    -   **Strengths:** Comprehensive in linking genes to functions through networks of interactions; useful for understanding high-level functions and utilities of the biological system.

3.  **Reactome**

    -   **Focus:** A curated database of pathways and reactions in human biology. Reactome pathways encompass various processes such as metabolism, signal transduction, gene expression, and cell cycle. It aims to provide a detailed, visual representation of molecular processes.

    -   **Strengths:** Highly detailed and manually curated; provides an interactive pathway browser that facilitates the exploration of complex biological pathways; also includes pathways from other species, providing a comparative perspective.

4.  **BioCarta**

    -   **Focus:** Earlier widely used for providing graphical representations of molecular pathways, focusing on various biological and signaling pathways. BioCarta's pathways were known for highlighting the interactions between different molecules and the potential regulation mechanisms.

    -   **Strengths:** Although less updated and less comprehensive than other databases mentioned, BioCarta pathways have been historically significant for offering insights into specific signaling and molecular pathways, contributing to the understanding of complex biological processes.

## Visualization of GSEA

#### Setup

```{r}
library(here)
library(readxl)
library(tidyverse)
library(patchwork)
```

### IL-1B 10 ng

#### KEGG

Testing just KEGG. Getting the code to work for one file.

```{r}
path_raw_data <- here("notebooks/seq/data_raw/GSEA/")
gsea_data <- read_excel(here(path_raw_data, "KEGG_IL1B_10_vs_vehicle.xlsx"))
```

```{r}
plot_data <- 
  arrange(gsea_data, -abs(NES)) %>% 
    group_by(sign(NES)) %>%
    slice(1:20)
```

```{r}
ggplot(data = plot_data,
       aes(x = NES,
           y = fct_reorder(pathway, NES),
           fill = padj
           )
       ) +
  geom_bar(stat = "identity") +
  scale_fill_distiller(type = "seq", palette = "YlOrRd")
  #scale_fill_continuous(low='red', high='blue', guide=guide_colorbar(reverse=TRUE), ) 
```

```{r}
ggplot(plot_data, aes(x = NES, y = fct_reorder(pathway, NES), color = padj, size = size)) + 
  geom_point(stat = 'identity') + 
  xlab("ratio") + ylab("path") + ggtitle("your data") + 
  theme_bw() + 
  scale_color_viridis_c() +
  #scale_color_distiller(type = "seq", palette = "PuRd") +
  geom_vline(xintercept = 0)
```


### All KEGG plots

Plot all using a function

```{r}
# Get file paths
kegg_file_paths <- dir(here("notebooks/seq/data_raw/GSEA"), pattern = "KEGG", full.names = TRUE)

# Subset to interleukines if wanted
#kegg_file_paths[grep(pattern = "IL", kegg_file_paths)]
```

```{r}
KEGG_plot_list <- map(kegg_file_paths, function(file_path) {
  # Read the file
  plot_data <- read_excel(file_path)
  
  # Process the data
  processed_data <- plot_data %>%
    arrange(desc(abs(NES))) %>%
    group_by(sign_NES = sign(NES)) %>%
    slice_head(n = 10) %>%
    ungroup() %>%
    filter(padj <= 0.05) %>%
    mutate(pathway = str_remove_all(pathway, "KEGG_"),
           pathway = str_replace_all(pathway, pattern = "_", replacement = " "),
           pathway = str_to_title(pathway))

  
  # Generate the plot
  ggplot(processed_data,
         aes(x = NES,
             y = fct_reorder(pathway, NES),
             color = padj,
             size = size
             )
         ) +
    geom_point(stat = 'identity') +
    labs(x = "NES",
         y = "Pathway",
         title = paste("Top 10 up/down Pathways for","", basename(file_path))
         ) +
    theme_bw() +
    geom_vline(xintercept = 0) +
    scale_color_viridis_c()
    #scale_color_distiller(type = "seq", palette = "YlOrRd") #+
    #theme(axis.text.y = element_text(size = 8))
})

```

```{r}
#| label: fig-kegg-abe

KEGG_plot_list[[1]]
```

```{r}
#| label: fig-kegg-il1b-0.1

KEGG_plot_list[[2]]
```

```{r}
#| label: fig-kegg-il1b-10

KEGG_plot_list[[3]]
```

```{r}
#| label: fig-kegg-il6-0.1

KEGG_plot_list[[4]]
```

```{r}
#| label: fig-kegg-il6-10

KEGG_plot_list[[5]]
```

```{r}
#| label: fig-kegg-il6-combined

KEGG_plot_list[[6]]
```

```{r}
#| label: fig-kegg-medium

KEGG_plot_list[[7]]
```

```{r}
#| label: fig-kegg-rif

KEGG_plot_list[[8]]
```

#### ALL BIOCARTA

**Plot all using a function**

Get all file paths related to BIOCARTA

```{r}
# Get file paths
biocarta_file_paths <- dir(here("notebooks/seq/data_raw/GSEA"), pattern = "BIOCARTA", full.names = TRUE)

# Subset to interleukines if wanted
# kegg_file_paths[grep(pattern = "IL", kegg_file_paths)]
```

```{r}
BIOCARTA_plot_list <- map(biocarta_file_paths, function(file_path) {
  # Read the file
  plot_data <- read_excel(file_path)
  
  # Process the data
  processed_data <- plot_data %>%
    arrange(desc(abs(NES))) %>%
    group_by(sign_NES = sign(NES)) %>%
    slice_head(n = 10) %>%
    ungroup() %>%
    filter(padj <= 0.05) %>%
    mutate(pathway = str_remove_all(pathway, "BIOCARTA"),
           pathway = str_replace_all(pathway, pattern = "_", replacement = " "),
           pathway = str_to_title(pathway))
  
  # Generate the plot
  ggplot(processed_data, aes(x = NES, y = fct_reorder(pathway, NES), color = padj, size = size)) +
    geom_point(stat = 'identity') +
    labs(x = "NES", y = "Pathway", title = paste("Top 10 up/down Pathways for","", basename(file_path))) +
    theme_bw() +
    geom_vline(xintercept = 0) +
    scale_color_viridis_c()
    #scale_color_distiller(type = "seq", palette = "YlOrRd") #+
    #theme(legend.position = "none") # Remove legend to focus on the data
})
```

```{r}
#| label: fig-biocarta-abe
BIOCARTA_plot_list[[1]]
```

```{r}
#| label: fig-biocarta-il1b-0.1
BIOCARTA_plot_list[[2]]
```

```{r}
#| label: fig-biocarta-il1b-10
BIOCARTA_plot_list[[3]]
```

```{r}
#| label: fig-biocarta-il6-0.1
BIOCARTA_plot_list[[4]]
```

```{r}
#| label: fig-biocarta-il6-10
BIOCARTA_plot_list[[5]]
```

```{r}
#| label: fig-biocarta-il6-combined
BIOCARTA_plot_list[[6]]
```

```{r}
#| label: fig-biocarta-medium
BIOCARTA_plot_list[[7]]
```

```{r}
#| label: fig-biocarta-rif
BIOCARTA_plot_list[[8]]
```

### REACTOME

Get all file paths related to Reactome

```{r}
# Get file paths
reactome_file_paths <- dir(here("notebooks/seq/data_raw/GSEA"), pattern = "REACTOME", full.names = TRUE)

# Subset to interleukines if wanted
# kegg_file_paths[grep(pattern = "IL", kegg_file_paths)]
```

```{r}
REACTOME_plot_list <- map(reactome_file_paths, function(file_path) {
  # Read the file
  plot_data <- read_excel(file_path)
  
  # Process the data
  processed_data <- plot_data %>%
    arrange(desc(abs(NES))) %>%
    group_by(sign_NES = sign(NES)) %>%
    slice_head(n = 10) %>%
    ungroup() %>%
    filter(padj <= 0.05) %>%
    mutate(pathway = str_remove_all(pathway, "REACTOME_"),
           pathway = str_replace_all(pathway, pattern = "_", replacement = " "),
           pathway = str_to_title(pathway))
  
  # Generate the plot
  ggplot(processed_data,
         aes(x = NES,
             y = fct_reorder(pathway, NES),
             color = padj,
             size = size
             )
         ) +
    geom_point(stat = 'identity') +
    labs(x = "NES",
         y = "Pathway",
         title = paste("Top 10 up/down Pathways for","", basename(file_path))
         ) +
    theme_bw() +
    geom_vline(xintercept = 0) +
    scale_color_viridis_c() 
    #scale_color_distiller(type = "seq", palette = "YlOrRd") #+
    #theme(legend.position = "none") # Remove legend to focus on the data
})
```

```{r}
#| label: fig-reactome-abe

REACTOME_plot_list[[1]]
```

```{r}
#| label: fig-reactome-il1b-0.1

REACTOME_plot_list[[2]]
```

```{r}
#| label: fig-reactome-il1b-10

REACTOME_plot_list[[3]]
```

```{r}
#| label: fig-reactome-il6-0.1

REACTOME_plot_list[[4]]
```

```{r}
#| label: fig-reactome-il6-10

REACTOME_plot_list[[5]]
```

```{r}
#| label: fig-reactome-il6-combined

REACTOME_plot_list[[6]]
```

```{r}
#| label: fig-reactome-medium

REACTOME_plot_list[[7]]
```

```{r}
#| label: fig-reactome-rif

REACTOME_plot_list[[8]]
```

### GO combined databases

CONSIDER :Split data into each of the GO databases:

1.  **Biological Process (BP)**, which describes the biological objectives to which the gene or gene product contributes.
2.  **Molecular Function (MF**), which describes the elemental activities of a gene product at the molecular level, such as binding or catalysis.
3.  **Cellular Component (CC)**, which describes where in the cell the gene product is active.

Get all file paths related to GO

```{r}
# Get file paths
go_file_paths <- dir(here("notebooks/seq/data_raw/GSEA"), pattern = "GO", full.names = TRUE)

# Subset to interleukines if wanted
# kegg_file_paths[grep(pattern = "IL", kegg_file_paths)]
```

```{r}
GO_plot_list <- map(go_file_paths, function(file_path) {
  # Read the file
  plot_data <- read_excel(file_path)
  
  # Process the data
  processed_data <- plot_data %>%
    arrange(desc(abs(NES))) %>%
    group_by(sign_NES = sign(NES)) %>%
    slice_head(n = 10) %>%
    ungroup() %>%
    filter(padj <= 0.05)
  
  # Generate the plot
  ggplot(processed_data,
         aes(x = NES,
             y = fct_reorder(pathway, NES),
             color = padj,
             size = size
             )
         ) +
    geom_point(stat = 'identity') +
    labs(x = "NES",
         y = "Pathway",
         title = paste("Top 10 up/down Pathways for","", basename(file_path))
         ) +
    theme_bw() +
    geom_vline(xintercept = 0) +
    scale_color_viridis_c() 
    #scale_color_viridis_c(option = "A")
    #scale_color_distiller(type = "seq", palette = "YlOrRd") #+
    #theme(legend.position = "none") # Remove legend to focus on the data
})
```

```{r}
#| label: fig-go-abe

GO_plot_list[[1]]
```

```{r}
#| label: fig-go-il1b-0.1

GO_plot_list[[2]]
```

```{r}
#| label: fig-go-il1b-10

GO_plot_list[[3]]
```

```{r}
#| label: fig-go-il6-0.1

GO_plot_list[[4]]
```

```{r}
#| label: fig-go-il6-10


GO_plot_list[[5]]
```

```{r}
#| label: fig-go-plots6
#| fig-width: 12
#| fig-height: 10


GO_plot_list[[6]]
```

```{r}
#| label: fig-go-medium
#| fig-width: 12
#| fig-height: 10

GO_plot_list[[7]]

```

```{r}
#| label: fig-go-rif
#| fig-width: 12
#| fig-height: 10

GO_plot_list[[8]]
```

### GO split databases

```{r}
# Read the GO file paths, and for each dataframe, add a column denoting 
# Which GO database was used.
GO_sorted_dfs <- map(go_file_paths, function(path) {
  read_excel(path) %>%
    mutate(db = case_when(
      str_detect(pathway, "GOBP") ~ "BP",
      str_detect(pathway, "GOCC") ~ "CC",
      str_detect(pathway, "GOMF") ~ "MF"
    ))
})

# Process the dataframes. Arrange by absolute NES vaule, group on positive or 
# negative regulation. Take just top ten of each group, and filter in only stat
# significant pathways.
GO_processed_data <- map(GO_sorted_dfs, function(df) {
    df %>%
    arrange(desc(abs(NES))) %>%
    group_by(sign_NES = sign(NES)) %>%
    slice_head(n = 10) %>%
    ungroup() %>%
    filter(padj <= 0.05)
})


# Function to generate plots for each db category within a single dataframe
generate_plots <- function(df, file_path) {
  db_categories <- unique(df$db)
  plots <- map(db_categories, function(db_category) {
    filtered_data <- filter(df, db == db_category)
    plot <- ggplot(filtered_data,
                   aes(x = NES,
                       y = fct_reorder(pathway, NES),
                       color = padj,
                       size = size)) +
      geom_point(stat = 'identity') +
      labs(x = "NES",
           y = "Pathway",
           title = paste("Top 10 up/down Pathways for", db_category, basename(file_path))) +
      theme_bw() +
      geom_vline(xintercept = 0) +
      scale_color_viridis_c()
    plot
  })
  names(plots) <- db_categories
  return(plots)
}

# Map over GO_processed_data to generate plots for each dataframe
GO_plots <- map2(GO_processed_data, go_file_paths, generate_plots)

# GO_plots is a list of lists of plots, with each inner list containing three plots (one for each db category)


# Extracting treatment names from the file paths
# Assuming the treatment names are directly before "vs_vehicle.xlsx" in the file names
treatment_names <- sapply(go_file_paths, function(path) {
  # Extract the part of the filename before "vs_vehicle.xlsx"
  treatment_name <- sub(".*GO_(.*)_vs_vehicle\\.xlsx$", "\\1", basename(path))
  # Replace underscores with spaces or any other necessary formatting
  treatment_name <- gsub("_", " ", treatment_name)
  treatment_name
})

# Now, apply these names to your combined_GO_plots list
names(GO_plots) <- treatment_names

```

```{r}
#| label: fig-go-split-il1b-0.1

# Combine plots from each database
combined_plot <- GO_plots[[2]]$BP + GO_plots[[2]]$CC + GO_plots[[2]]$MF + 
  plot_layout(ncol = 1) # Arranges the plots in a single column

# Print the combined plot
print(combined_plot)
```

```{r}
#| label: fig-go-split-il1b-10
#| fig-width: 20


# Combine plots from each database
combined_plot <- GO_plots[[3]]$BP + GO_plots[[3]]$CC + GO_plots[[3]]$MF + 
  plot_layout(ncol = 1) # Arranges the plots in a single column

# Print the combined plot
print(combined_plot)
```

```{r}
#| label: fig-go-split-i6-0.1

# Combine plots from each database
combined_plot <- GO_plots[[4]]$BP + GO_plots[[4]]$CC + GO_plots[[4]]$MF + 
  plot_layout(ncol = 1) # Arranges the plots in a single column

# Print the combined plot
print(combined_plot)
```

```{r}
#| label: fig-go-split-i6-10

# Combine plots from each database
combined_plot <- GO_plots[[5]]$BP + GO_plots[[5]]$CC + GO_plots[[5]]$MF + 
  plot_layout(ncol = 1) # Arranges the plots in a single column

# Print the combined plot
print(combined_plot)
```

```{r}
#| label: fig-go-split-i6-combined

# Combine plots from each database
combined_plot <- GO_plots[[6]]$BP + GO_plots[[6]]$CC + GO_plots[[6]]$MF + 
  plot_layout(ncol = 1) # Arranges the plots in a single column

# Print the combined plot
print(combined_plot)
```

```{r}
#| label: fig-go-split-medium

# Combine plots from each database
combined_plot <- GO_plots[[7]]$BP + GO_plots[[7]]$CC + GO_plots[[7]]$MF + 
  plot_layout(ncol = 1) # Arranges the plots in a single column

# Print the combined plot
print(combined_plot)
```

